### PR TITLE
Count recording rules used in rules group as used

### PR DESCRIPTION
**Pull request by vaxvms**
#### What this PR does

Recording rules used in the same rules group are not marked as used.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.

### CODE REVIEW

**Comment by dimitarvdimitrov** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Review by dimitarvdimitrov — COMMENTED** [review]
I don't see anything wrong with the change. However, it is a behaviour change. So at the least, the changelog entry should say `CHANGE` instead of `BUGFIX`.

I know @wilfriedroset is relying on the exported mimirtool functions. @wilfriedroset would this change break you guys?

The original PR which introduced removing rules from the set of used metric names is also the PR which introduced the `analyze ruler` command - [see here](https://github.com/grafana/cortex-tools/pull/201/files#diff-491cc1d09479ab4265b59602f3f5a8a257d7b5e3aebeea6e7683829975d3da18R87-R90) - and doesn't contain extra info.

**Comment by wilfriedroset** [comment]
all good on my end we have tested with @vaxvms that the behavior is the one expected.

File: CHANGELOG.md [hunk 1]
```diff
@@ -237,6 +237,7 @@
 * [BUGFIX] `mimirtool rules prepare`: do not add aggregation label to `on()` clause if already present in `group_left()` or `group_right()`. #7839
 * [BUGFIX] Analyze Grafana: fix parsing queries with variables. #8062
 * [BUGFIX] `mimirtool rules sync`: detect a change when the `query_offset` or the deprecated `evaluation_delay` configuration changes. #8297
+* [CHANGE] Analyze Rules: Count recording rules used in rules group as used. #6133
```
Comment (wilfriedroset) [inline CHANGELOG.md:240]: move this line at just after the last `[CHANGE]` entry
cf: https://github.com/grafana/mimir/tree/main/docs/internal/contributing#changelog

**Review by wilfriedroset — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -229,6 +229,7 @@
 ### Mimirtool
 
 * [CHANGE] Deprecated `--rule-files` flag in favor of CLI arguments. #7756
+* [CHANGE] Analyze Rules: Count recording rules used in rules group as used. #6133
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:232]: can you move this entry to the unreleased section on top of the file?

**Review by dimitarvdimitrov — APPROVED** [review]
SGTM, thank you. Only a note about the changelog entry

**Comment by dimitarvdimitrov** [comment]
the changelog entry still looks like it's in the 2.13 section

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thanks! :)

**Pull request merged**